### PR TITLE
Added functions to monitor and control SQLite memory usage

### PR DESCRIPTION
**Pull request by anton-dessiatov**
Also updated changelog and made a version bump in .cabal file.

No changes to existing API were made, just added 2 data structures and 2 functions. This stuff served a good job for me debugging application memory usage so it might be useful to others.

### CODE REVIEW

File: persistent-sqlite/Database/Sqlite.hs [hunk 1]
```diff
@@ -532,4 +536,101 @@ config c = case c of
...
+  -- | This parameter records the largest memory allocation request handed to
+  -- scratch memory allocator. Only the value returned in the 'sqliteStatusHighwater'
+  -- field of 'SqliteStatus' record is of interest. The value written into the
+  -- 'sqliteStatusCurrent' field is Nothing.
+  | SqliteStatusScratchSize
+  -- | This parameter records the number of separate memory allocations currently
+  -- checked out.
+  | SqliteStatusMallocCount
+
+statusVerbInfo :: SqliteStatusVerb -> (CInt, Bool, Bool)
```
Comment (snoyberg) [inline persistent-sqlite/Database/Sqlite.hs]: Could we have a comment explaining what these values mean?

File: persistent-sqlite/ChangeLog.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 2.6.0.2
```
Comment (snoyberg) [inline persistent-sqlite/ChangeLog.md]: With API extension, this should be a minor version bump (2.6.1).

File: persistent-sqlite/Database/Sqlite.hs [hunk 3]
```diff
@@ -532,4 +536,101 @@ config c = case c of
       ErrorOK -> return ()
       _ -> sqlError Nothing "sqlite3_config" e
 
-
+-- | Return type of the 'status' function
+--
+-- Since 2.6.0.2
```
Comment (snoyberg) [inline persistent-sqlite/Database/Sqlite.hs]: Can you switch this to `@since 2.6.1` (and elsewhere)?

File: persistent-sqlite/cbits/config.c [hunk 4]
```diff
@@ -4,3 +4,7 @@
 int persistent_sqlite_set_log(void (*logFn)(void*, int, const char*), void* arg) {
   return sqlite3_config(SQLITE_CONFIG_LOG, logFn, arg);
 }
+
+long long int persistent_sqlite3_soft_heap_limit64(long long int N) {
```
Comment (snoyberg) [inline persistent-sqlite/cbits/config.c]: What is this function adding versus just calling `sqlite3_soft_heap_limit64` directly?

File: persistent-sqlite/ChangeLog.md (hunk 2 above)
Comment (snoyberg) [inline persistent-sqlite/ChangeLog.md]: Can you include a bump in the .cabal file?

File: persistent-sqlite/cbits/config.c (hunk 4 above)
Comment (anton-dessiatov) [inline persistent-sqlite/cbits/config.c]: It's just a layer of protection from future versions of SQlite doing something weird with sqlite3_int64 type. Currently that type is just a typedef for long long it so this function really does nothing. If, however, SQLite ever changes sqlite3_int64 to something different than long long int then this function will either do implicit conversion (if we are sort of lucky) or at least gives us a C compiler error.

**Comment by snoyberg** [comment]
I think it's better if we drop it here. SQLite has shown a remarkable level
of API stability, I'm not worried about them suddenly changing that.

**Comment by snoyberg** [comment]
Code looks good, just a few Travis failures that need to be addressed, e.g.:

```
Database/Sqlite.hs:634:74: Not in scope: ‘pure’
Database/Sqlite.hs:635:7:
    Not in scope: ‘<*>’
    Perhaps you meant ‘<$>’ (imported from Control.Applicative)
```

https://travis-ci.org/yesodweb/persistent/jobs/205780522

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**